### PR TITLE
Feature/BISC-86 create student learning need

### PR DESCRIPTION
**Pull request by dirkkok**
This PR will:
- add `createLearningNeed` mutation
- add `learningNeeds` query

The code for the REST calls isn't very pretty, but we'll figure that out later. Most important is that our internal graphql API is ready.

### CODE REVIEW

File: server/src/LearningNeed/services/LearningNeedService.ts [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+        const desiredOutComesTopicOther = bodyJson.gewensteleeruitkomstonderwerpanders
+        const desiredOutComesApplication = bodyJson.gewensteleeruitkomsttoepassing as LearningNeedApplicationEnum
+        const desiredOutComesApplicationOther = bodyJson.gewensteleeruitkomsttoepassinganders
+        const desiredOutComesLevel = bodyJson.gewensteleeruitkomstniveau as LearningNeedLevelEnum
+        const desiredOutComesLevelOther = bodyJson.gewensteleeruitkomstniveauanders
+        const offerDesiredOffer = bodyJson.aanbodgewensteaanbod
+        const offerAdvisedOffer = bodyJson.aanbodgeadviseerdaanbod
+        const offerDifference = bodyJson.aanbodverschilwensadvies as LearningNeedOfferDifferenceEnum
+        const offerDifferenceOther = bodyJson.aanbodverschilwensadviesanders
+        const offerEngagements = bodyJson.aanbodafspraken
```
Comment (Jorann) [inline server/src/LearningNeed/services/LearningNeedService.ts:151]: So ugly  😢

File: server/src/LearningNeed/services/LearningNeedService.ts [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+                    method: 'GET',
+                    body,
+                    headers: {
+                        'Content-Type': 'application/json',
+                        Authorization: this.configService.get('API_KEY'),
+                    },
+                },
+                (err, res) => {
+                    this.logger.debug(body)
+                    this.logger.debug(res.body)
```
Comment (Jorann) [inline server/src/LearningNeed/services/LearningNeedService.ts:192]: debugs?

File: server/src/LearningNeed/services/LearningNeedService.ts [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+                    if (err) {
+                        reject(err)
+                        return
+                    }
+
+                    // 200 = ok
+                    if (res.statusCode === 200) {
```
Comment (Jorann) [inline server/src/LearningNeed/services/LearningNeedService.ts:200]: No 201 here?

File: server/src/LearningNeed/types/LearningNeedType.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    public desiredOutComesTopic!: LearningNeedTopicEnum
+
+    @Field(() => String, { nullable: true })
+    public desiredOutComesTopicOther!: string | null
+
+    @Field(() => LearningNeedApplicationEnum)
+    public desiredOutComesApplication!: LearningNeedApplicationEnum
+
+    @Field(() => String, { nullable: true })
+    public desiredOutComesApplicationOther!: string | null
```
Comment (Jorann) [inline server/src/LearningNeed/types/LearningNeedType.ts:33]: needs a ? not a !

File: server/src/LearningNeed/types/LearningNeedType.ts [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    public learningNeedMotivation!: string
+
+    @Field()
+    public desiredOutComesGoal!: string
+
+    @Field(() => LearningNeedTopicEnum)
+    public desiredOutComesTopic!: LearningNeedTopicEnum
+
+    @Field(() => String, { nullable: true })
+    public desiredOutComesTopicOther!: string | null
```
Comment (Jorann) [inline server/src/LearningNeed/types/LearningNeedType.ts:27]: needs a ? not a !

File: server/src/LearningNeed/types/LearningNeedType.ts [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    public desiredOutComesApplication!: LearningNeedApplicationEnum
+
+    @Field(() => String, { nullable: true })
+    public desiredOutComesApplicationOther!: string | null
+
+    @Field(() => LearningNeedLevelEnum)
+    public desiredOutComesLevel!: LearningNeedLevelEnum
+
+    @Field(() => String, { nullable: true })
+    public desiredOutComesLevelOther!: string | null
```
Comment (Jorann) [inline server/src/LearningNeed/types/LearningNeedType.ts:39]: needs a ? not a !

File: server/src/LearningNeed/types/LearningNeedType.ts [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+    public offerDesiredOffer!: string
+
+    @Field()
+    public offerAdvisedOffer!: string
+
+    @Field(() => LearningNeedOfferDifferenceEnum)
+    public offerDifference!: LearningNeedOfferDifferenceEnum
+
+    @Field(() => String, { nullable: true })
+    public offerDifferenceOther!: string | null
```
Comment (Jorann) [inline server/src/LearningNeed/types/LearningNeedType.ts:51]: needs a ? not a !

File: server/src/LearningNeed/types/LearningNeedType.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+    @Field(() => String, { nullable: true })
+    public offerEngagements!: string | null
```
Comment (Jorann) [inline server/src/LearningNeed/types/LearningNeedType.ts:54]: needs a ? not a !

**Review by Jorann — CHANGES_REQUESTED** [review]

File: server/src/LearningNeed/services/LearningNeedService.ts (hunk 3 above)
Comment (dirkkok) [reply server/src/LearningNeed/services/LearningNeedService.ts:200]: Create returns 201 on success, get returns 200 on success

**Pull request merged**